lib/engine/game/g_1862/step/option_share.rb

```diff
@@ -0,0 +1,91 @@
+# frozen_string_literal: true
+
+require_relative '../../../step/base'
+
+module Engine
+  module Game
+    module G1862
+      module Step
+        class OptionShare < Engine::Step::Base
+          def actions(entity)
+            return [] unless entity == pending_entity
+
+            ['choose']
+          end
+
+          def description
+            'Option share choice'
+          end
+
+          def active_entities
+            [pending_entity]
+          end
+
+          def active?
+            pending_entity
+          end
+
+          def pending_entity
+            pending_option[:entity]
+          end
+
+          def pending_option
+            @round.pending_options&.first || {}
+          end
+
+          def process_choose(action)
+            entity = action.entity
+            share = pending_option[:share]
+            if action.choice == :sell
+              @game.bank.spend(pending_option[:sell_price], entity)
+              @game.transfer_share(share, @game.share_pool)
+              @log << "#{entity.name} sells #{pending_option[:share].corporation.name} #{pending_option[:percent]}% "\
+                "option share for #{@game.format_currency(pending_option[:sell_price])}"
+            else
+              entity.spend(pending_option[:redeem_price], @game.bank)
+              @log << "#{entity.name} redeems #{pending_option[:share].corporation.name} #{pending_option[:percent]}% "\
+                "option share for #{@game.format_currency(pending_option[:redeem_price])}"
+            end
+
+            @round.pending_options.shift
+            @game.continue_merge_option(entity)
+          end
+
+          def choice_name
+            if pending_option[:entity].corporation?
+              'Sell or Buy option (partial) share on behalf of surviving company'
+            elsif pending_option[:percent] > 10
+              'Sell or Buy option (partial) share - may allow retaining presidency'
+            else
+              'Sell or Buy option (partial) share'
+            end
+          end
+
+          def choices
+            if pending_option[:percent] > 10
+              {
+                sell: "Sell #{pending_option[:share].corporation.name} #{pending_option[:percent]}% option "\
```

Comment:
Reuse code?

director_text = pending_option[:percent] > 10  ? ' (Director's certificate)' : ''
"share#{director_text} for ..."

Reply:
Sure.